warehouse/query-core/src/test/java/datawave/query/LongRunningQueryTest.java

```diff
@@ -0,0 +1,140 @@
+package datawave.query;
+
+import com.google.common.collect.Sets;
+import datawave.common.test.integration.IntegrationTest;
+import datawave.helpers.PrintUtility;
+import datawave.marking.MarkingFunctions;
+import datawave.query.tables.ShardQueryLogic;
+import datawave.query.util.DateIndexHelperFactory;
+import datawave.query.util.MetadataHelperFactory;
+import datawave.query.util.VisibilityWiseGuysIngest;
+import datawave.security.authorization.DatawavePrincipal;
+import datawave.security.authorization.DatawaveUser;
+import datawave.security.authorization.SubjectIssuerDNPair;
+import datawave.security.util.DnUtils;
+import datawave.util.TableName;
+import datawave.webservice.common.connection.AccumuloConnectionFactory;
+import datawave.webservice.query.QueryImpl;
+import datawave.webservice.query.cache.QueryMetricFactoryImpl;
+import datawave.webservice.query.cache.ResultsPage;
+import datawave.webservice.query.configuration.GenericQueryConfiguration;
+import datawave.webservice.query.logic.AbstractQueryLogicTransformer;
+import datawave.webservice.query.result.event.DefaultResponseObjectFactory;
+import datawave.webservice.query.runner.RunningQuery;
+import org.apache.accumulo.core.client.Connector;
+import org.apache.accumulo.core.security.Authorizations;
+import org.apache.commons.collections4.iterators.TransformIterator;
+import org.apache.log4j.Logger;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import java.text.DateFormat;
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+import static org.junit.Assert.assertEquals;
+
+public class LongRunningQueryTest {
```

Comment:
Good stuff!  How about a few more tests:
1. query for something legitimately nonexistent and make sure it returns a 204 successfully
2. set the pageSize to something less than the number of results and make sure nothing goes awry there

Reply:
Added `testAllowLongRunningQueryWithSmallPageSize()` and `testAllowLongRunningQueryOnNoResultsc()`